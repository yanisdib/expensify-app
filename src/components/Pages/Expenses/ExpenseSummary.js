import React from 'react';
import numeral from 'numeral';
import '../../../numeral/formats/euro';
import { connect } from 'react-redux';
import selectExpenses from '../../../selectors/expenses';
import getExpensesTotal from '../../../selectors/expenses-total';

export const ExpenseSummary = ({ expensesCount, expensesTotal }) => {
    const expenseWord = expensesCount <= 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal).format('0,0.00â‚¬').replace(',',' ');
    return (
        <div>
            <h5>View {expensesCount} {expenseWord} totalling {formattedExpensesTotal}</h5>
        </div>
    );
};
const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);
