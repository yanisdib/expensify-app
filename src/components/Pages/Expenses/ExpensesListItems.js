import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItems = ({ id, description, amount, createdAt, note }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>{numeral(amount).format('$0.00')} - {moment(createdAt).format('DD/MM/YYYY')}</p>
        {note ? (
            <p>Note: {note}</p>
        ) : (
                null
            )}
    </div>
);

export default ExpenseListItems;
