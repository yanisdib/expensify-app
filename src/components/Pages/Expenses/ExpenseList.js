import React from 'react';
import { connect } from 'react-redux';
import ExpensesListItems from './ExpensesListItems';
import selectExpenses from '../../../selectors/expenses';


export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length !== 0 ? (
                props.expenses.map((expense) => {
                    return <ExpensesListItems key={expense.id} {...expense} />
                })
            ) : (
                    <p>No expenses...</p>
                )
        }
    </div>
);
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};
// What we want to connect
// Function - Define what info we want to access from the store
export default connect(mapStateToProps)(ExpenseList);