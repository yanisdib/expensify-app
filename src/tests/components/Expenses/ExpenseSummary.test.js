import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../../components/Pages/Expenses/ExpenseSummary';

test('Should correctly render ExpenseSummary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={12.94} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={4} expenseTotal={12458} />);
    expect(wrapper).toMatchSnapshot();
});