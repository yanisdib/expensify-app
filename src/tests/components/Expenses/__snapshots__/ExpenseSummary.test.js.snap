// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h3>
    View 
     
    expenses
     totalling 
    $00.00
  </h3>
</div>
`;

exports[`Should correctly render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h3>
    View 
     
    expenses
     totalling 
    $00.00
  </h3>
</div>
`;
